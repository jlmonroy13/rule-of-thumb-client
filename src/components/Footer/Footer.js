import React from 'react';
import { NavLink } from 'react-router-dom';

import Icon from 'components/Icon';

import {
  TERMS_AND_CONDITIONS,
  PRIVACY_POLICY,
  CONTACT_US,
} from 'constants/pathNames';

const Footer = () => (
  <div className="container">
    <div className="footer">
      <nav className="footer__nav">
        <NavLink to={TERMS_AND_CONDITIONS} className="footer__nav-item">
          Terms and Conditions
        </NavLink>
        <NavLink to={PRIVACY_POLICY} className="footer__nav-item">
          Privacy Policy
        </NavLink>
        <NavLink to={CONTACT_US} className="footer__nav-item">
          Contact us
        </NavLink>
      </nav>
      <div className="footer__social-networks">
        <span className="footer__text">Follow Us</span>
        <a
          rel="noopener noreferrer"
          href="https://www.facebook.com/pages/category/Community/FB-Rules-of-Thumb-318882558219309/"
          target="_blank"
        >
          <Icon
            name="facebook"
            className="footer__icon footer__icon--facebook"
          />
        </a>
        <a
          rel="noopener noreferrer"
          href="https://twitter.com/ruleofthumbbiz"
          target="_blank"
        >
          <Icon name="twitter" className="footer__icon footer__icon--twitter" />
        </a>
      </div>
    </div>
  </div>
);

export default Footer;
