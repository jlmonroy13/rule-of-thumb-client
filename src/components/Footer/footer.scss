.footer {
  display: flex;
  justify-content: space-between;
  border-top: 1px dotted $body-color;
  padding: 25px 0;

  &__nav {
    width: 40%;
    display: flex;
    justify-content: space-between;
    align-items: center;

    &-item {
      @include font-size(13);

      color: $dove-gray;
      font-weight: 300;

      &:hover {
        color: $dove-gray;
      }

      @include media-breakpoint-down(sm) {
        text-align: center;
      }
    }

    @include media-breakpoint-down(sm) {
      width: unset;
      padding-right: 40px;
      flex-grow: 1;
    }
  }

  &__icon {
    @include font-size(32);

    color: $mine-shaft;
    transition: color 0.2s;

    &--facebook {
      &:hover {
        color: $chambray;
      }
    }

    &--twitter {
      &:hover {
        color: $cerulean;
      }
    }
  }

  &__text {
    @include font-size(13);

    color: $dove-gray;
    font-weight: 300;

    @include media-breakpoint-down(sm) {
      display: none;
    }
  }

  &__social-networks {
    width: 170px;
    display: flex;
    justify-content: space-between;
    align-items: center;

    @include media-breakpoint-down(sm) {
      width: 80px;
    }
  }
}
