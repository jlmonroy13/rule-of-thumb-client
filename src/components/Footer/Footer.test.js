/**
 *
 * Tests for Footer
 *
 * @see https://github.com/react-boilerplate/react-boilerplate/tree/master/docs/testing
 *
 */

import React from 'react';
import { render } from '@testing-library/react';
import { BrowserRouter as Router } from 'react-router-dom';

import Footer from './index';

const props = {};

describe('<Footer />', () => {
  it('Expect to not log errors in console', () => {
    const spy = jest.spyOn(global.console, 'error');
    render(
      <Router>
        <Footer {...props} />
      </Router>,
    );
    expect(spy).not.toHaveBeenCalled();
  });

  it('Should render and match the snapshot', () => {
    const {
      container: { firstChild },
    } = render(
      <Router>
        <Footer {...props} />
      </Router>,
    );
    expect(firstChild).toMatchSnapshot();
  });
});
