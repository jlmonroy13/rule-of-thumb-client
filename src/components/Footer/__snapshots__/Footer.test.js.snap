// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Footer /> Should render and match the snapshot 1`] = `
<div
  class="container"
>
  <div
    class="footer"
  >
    <nav
      class="footer__nav"
    >
      <a
        class="footer__nav-item"
        href="/terms-and-conditions"
      >
        Terms and Conditions
      </a>
      <a
        class="footer__nav-item"
        href="/privacy-policy"
      >
        Privacy Policy
      </a>
      <a
        class="footer__nav-item"
        href="/contact-us"
      >
        Contact us
      </a>
    </nav>
    <div
      class="footer__social-networks"
    >
      <span
        class="footer__text"
      >
        Follow Us
      </span>
      <a
        href="https://www.facebook.com/pages/category/Community/FB-Rules-of-Thumb-318882558219309/"
        rel="noopener noreferrer"
        target="_blank"
      >
        <svg
          class="icon icon-facebook footer__icon footer__icon--facebook"
          data-testid="icon-component"
        >
          <use
            xlink:href="#icon-facebook"
          />
        </svg>
      </a>
      <a
        href="https://twitter.com/ruleofthumbbiz"
        rel="noopener noreferrer"
        target="_blank"
      >
        <svg
          class="icon icon-twitter footer__icon footer__icon--twitter"
          data-testid="icon-component"
        >
          <use
            xlink:href="#icon-twitter"
          />
        </svg>
      </a>
    </div>
  </div>
</div>
`;
